core/class/luna.class.php

```diff
@@ -869,8 +869,88 @@
 
   /* ------ FIN 4G ----- */
 
+  /* ----- DEBUT SMS  ----- */
+
+  public static function listGlobalSMS(){
+    $list = self::listSMS();
+    $return = [];
+    foreach ($list as $sms) {
+      $sms = self::readSMS($sms);
+      $return[] = $sms;
+    }
+    return $return;
+  }
+
+  public static function listSMS() {
+    $list = shell_exec('sudo mmcli -m 0 --messaging-list-sms');
+    $list = explode("\n", $list);
+    $listSMS = [];
+    foreach ($list as $sms) {
+      if (preg_match('/\/org\/freedesktop\/ModemManager1\/SMS\/[0-9]+/', $sms)) {
+        $sms = explode('/', $sms);
+        $sms = $sms[5];
+        $sms = explode(' ', $sms);
+        $sms = $sms[0];
+        $listSMS[] = $sms;
+      }
+    }
+    return $listSMS;
+  }
+
+  public static function readSMS($sms) {
+    $sms = shell_exec('sudo mmcli -s ' . $sms);
+    $list = explode("\n", $sms);
+    $listSMS = [];
+    foreach ($list as $sms) {
+      if (preg_match('/Content    |    number:/', $sms)) {
+        $sms = explode(':', $sms);
+        $sms = $sms[1];
+        $sms = trim($sms);
+        $listSMS['number'] = $sms;
+      }
+      if (preg_match('/text:/', $sms)) {
+        $sms = explode(':', $sms);
+        $sms = $sms[1];
+        $sms = trim($sms);
+        $listSMS['text'] = $sms;
+      }
+      if (preg_match('/smsc:/', $sms)) {
+        $sms = explode(':', $sms);
+        $sms = $sms[1];
+        $sms = trim($sms);
+        $listSMS['smsc'] = $sms;
+      }
+      if (preg_match('/state:/', $sms)) {
+        $sms = explode(':', $sms);
+        $sms = $sms[1];
+        $sms = trim($sms);
+        $listSMS['state'] = $sms;
+      }
+    }
+    return $listSMS;
+  }
+
+  public static function deleteSMS($sms) {
+    shell_exec('sudo mmcli -s ' . $sms . ' --messaging-delete-sms=' . $sms);
+  }
+
+  public static function sendSMS($number, $text) {
```

Comment:
The sendSMS function could return a status or result of the operation, to let the caller know if the SMS was sent successfully or not. [medium]